### PR TITLE
Exit with an error code when generate command fails.

### PR DESCRIPTION
**Pull request by desistefanova**
Fix `dart run realm_dart generate` command to exit with an error code when generation fails.
Otherwise, it does not terminate the execution of the command batch.
The CI for generating models for the samples after the new realm is released didn't fail on generating error.
Fixes #956

### CODE REVIEW

File: lib/src/cli/generate/generate_command.dart [hunk 1]
```diff
@@ -42,10 +42,18 @@ class GenerateCommand extends Command<void> {
...
+          : options.watch
+              ? 'watch'
+              : 'build',
       ...[if (!options.clean) '--delete-conflicting-outputs'], // not legal option to clean
     ]);
-    
+
     await stdout.addStream(process.stdout);
+    var exitCode = await process.exitCode;
+    if (exitCode != 0) {
```
Comment (blagoev) [inline lib/src/cli/generate/generate_command.dart:55]:
```suggestion
   exit(exitCode);
```

File: lib/src/cli/generate/generate_command.dart [hunk 2]
```diff
@@ -42,10 +42,18 @@ class GenerateCommand extends Command<void> {
...
+      exit(exitCode);
```
Comment (blagoev) [inline lib/src/cli/generate/generate_command.dart:56]:
```suggestion
   
```

File: lib/src/cli/generate/generate_command.dart [hunk 3]
```diff
@@ -42,10 +42,18 @@ class GenerateCommand extends Command<void> {
...
+          ? 'clean'
+          : options.watch
+              ? 'watch'
+              : 'build',
       ...[if (!options.clean) '--delete-conflicting-outputs'], // not legal option to clean
     ]);
-    
+
     await stdout.addStream(process.stdout);
+    var exitCode = await process.exitCode;
```
Comment (blagoev) [inline lib/src/cli/generate/generate_command.dart:54]:
```suggestion
    final exitCode = await process.exitCode;
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
 ### Fixed
 * Added more validations when using `User.apiKeys` to return more meaningful errors when the user cannot perform API key actions - e.g. when the user has been logged in with API key credentials or when the user has been logged out. (Issue [#950](https://github.com/realm/realm-dart/issues/950))
+* Fixed `dart run realm_dart generate` and `flutter pub run realm generate` commands to exit with an error code when generation fails. Execution is now terminated when the command is run in a batch with other commands.
```
Comment (blagoev) [inline CHANGELOG.md:10]:
```suggestion
* Fixed `dart run realm_dart generate` and `flutter pub run realm generate` commands to exit with the correct error code when generation fails.
```

**Review by blagoev — APPROVED** [review]
Added some code and changelog suggestions

**Review by nirinchev — APPROVED** [review]

**Pull request merged**